### PR TITLE
Fixed subtitles being transcoded when they are in SupportedInternalSubtitlesFormats or SupportedExternalSubtitlesFormats

### PR DESCRIPTION
**Pull request by SubJunk**

### CODE REVIEW

File: src/main/java/net/pms/configuration/FormatConfiguration.java [hunk 1]
```diff
@@ -463,12 +478,22 @@ public boolean match(
 				if (isExternalSubs) {
 					if (supportedExternalSubtitlesFormats == null || !subsFormat.matches(supportedExternalSubtitlesFormats)) {
 						LOGGER.trace("External subtitles format \"{}\" failed to match support line {}", subsFormat, supportLine);
-						return false;
+						if (!renderer.isExternalSubtitlesFormatSupportedForAllFiletypes(subsFormat)) {
```
Comment (SubJunk) [inline src/main/java/net/pms/configuration/FormatConfiguration.java:481]:
```suggestion
						if (renderer == null || !renderer.isExternalSubtitlesFormatSupportedForAllFiletypes(subsFormat)) {
```

File: src/main/java/net/pms/configuration/FormatConfiguration.java [hunk 2]
```diff
@@ -463,12 +478,22 @@ public boolean match(
...
+							return false;
+						} else {
+							LOGGER.trace("But did match a format in the SupportedExternalSubtitlesFormats renderer configuration setting");
+						}
 					}
 				} else {
 					if (supportedEmbeddedSubtitlesFormats == null || !subsFormat.matches(supportedEmbeddedSubtitlesFormats)) {
 						LOGGER.trace("Internal subtitles format \"{}\" failed to match support line {}", subsFormat, supportLine);
-						return false;
+						if (!renderer.isEmbeddedSubtitlesFormatSupportedForAllFiletypes(subsFormat)) {
```
Comment (SubJunk) [inline src/main/java/net/pms/configuration/FormatConfiguration.java:491]:
```suggestion
						if (renderer == null || !renderer.isEmbeddedSubtitlesFormatSupportedForAllFiletypes(subsFormat)) {
```

**Comment by valib** [comment]
@SubJunk the check for format is repeated. See

```TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Matched external subtitles track for "20150214_182916_x264.mp4": External, type: SUBRIP, lang: und, externalFile: E:\Filmy\20150214_182916_x264.srt, external file character set: UTF-8
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Getting player for resource "20150214_182916_x264.mp4"
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Returning compatible player "FFmpeg Video"
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:3gp        v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000     m:video/3gpp
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:avi  	 	 v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000	   m:video/avi
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:flv        v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000     m:video/x-flv
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] External subtitles format "SUBRIP" failed to match support line f:mp4        v:h264|h265|mp4|mpeg1|mpeg2|vc1|vp6|wmv   		 a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000 	   m:video/mp4
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] But did match a format in the SupportedExternalSubtitlesFormats renderer configuration setting
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Matched support line f:mp4        v:h264|h265|mp4|mpeg1|mpeg2|vc1|vp6|wmv   		 a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000 	   m:video/mp4
DEBUG 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Final verdict: "20150214_182916_x264.mp4" will be streamed
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:3gp        v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000     m:video/3gpp
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:avi  	 	 v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000	   m:video/avi
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Format "mp4" failed to match supported line f:flv        v:h264|h265|mjpeg|mp4|mpeg1|mpeg2|vc1|vp6|wmv   a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000     m:video/x-flv
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] External subtitles format "SUBRIP" failed to match support line f:mp4        v:h264|h265|mp4|mpeg1|mpeg2|vc1|vp6|wmv   		 a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000 	   m:video/mp4
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] But did match a format in the SupportedExternalSubtitlesFormats renderer configuration setting
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] Matched support line f:mp4        v:h264|h265|mp4|mpeg1|mpeg2|vc1|vp6|wmv   		 a:aac-lc|he-aac|ac3|dts|eac3|lpcm|mp3|wma    w:1920 h:1080 b:20000000 	   m:video/mp4
TRACE 2020-12-21 10:39:16.747 [HTTPv2 Request Worker 6] File "20150214_182916_x264.mp4" will be sent with MIME type "video/mp4"
```

 After `Final verdict: "20150214_182916_x264.mp4" will be streamed` it checks the format again to recognize the MIME.

**Comment by SubJunk** [comment]
Isn't that the same in master? There are double checks but this should reduce them in some situations, not increase them

This fixes a bug I caused in https://github.com/UniversalMediaServer/UniversalMediaServer/pull/2301

**Comment by valib** [comment]
@SubJunk I know and I tried to solve it without success. Maybe this is the time to find where is the problem.

**Review by valib — APPROVED** [review]

**Comment by SubJunk** [comment]
Yeah there could still be reformatting. I think this change, which moves that logic into the central format support place, is going to be more stable. It's not perfect but it is an improvement, I hope :)

**Pull request merged**